lib/ecto/model/timestamps.ex

```diff
@@ -23,11 +23,23 @@ defmodule Ecto.Model.Timestamps do
     if get_change changeset, field do
       changeset
     else
-      {date, {h, m, s}} = :erlang.universaltime
-      put_change changeset, field, Ecto.Type.load!(type, {date, {h, m, s, 0}})
+      put_change changeset, field, Ecto.Type.load!(type, timestamp_tuple_with_usec(changeset.repo.adapter))
     end
   end
 
+  defp timestamp_tuple_with_usec(Ecto.Adapters.MySQL) do
```

Comment:
OK, this passes integration tests on Travis. Instead of looking at the name of the adapter, we could have a boolean function in adapters to determine whether there is consistent microsecond support. What would be a good name for it?

`supports_microsecond_datetimes?` ?